psi4/src/psi4/ccenergy/ccenergy.cc

```diff
@@ -331,6 +332,17 @@ double CCEnergyWavefunction::compute_energy()
 
             outfile->Printf( "\n");
             amp_write();
+
+			// Get T amplitudes
+            global_dpd_->file2_init(&t1, PSIF_CC_OEI, 0, 0, 1, "tIA");
+            SharedMatrix new_t1 = SharedMatrix(new Matrix(&t1));
+            set_T1(new_t1);
+            global_dpd_->file2_close(&t1);
+            global_dpd_->buf4_init(&t2, PSIF_CC_TAMPS, 0, 0, 5, 0, 5, 0, "tIjAb");
+            SharedMatrix new_t2 = SharedMatrix(new Matrix(&t2));
+            set_T2(new_t2);
+            global_dpd_->buf4_close(&t2);
```

Comment:
If I understand this correctly this will set T1/T2 in memory for every run which isn't feasible for large computations. This will need to be an option where a CCWavefunction optionally reads this into memory on demand.

Reply:
Hmm, how would you call such an option? Maybe it would also work to set this dynamically in memory, when it is accessed via the getter?

Reply:
You might want to consider an option does not delete T1/T2 and then a function that pulls this from memory `SharedMatrix get_T2()`. The CC guys could comment more.

Reply:
I agree with @dgasmith T1 would probably be fine, but T2 will be too large to store in core for many use cases.

Personally I think a good solution would be to leave the `T1_` and `T2_` `SharedMatrix` objects uninitialized until some function is explicitly called by the user to populate them. If the functions returning T1/T2 are called before T1/T2 are initialized they could throw, or populate at that time. 

To address @dgasmith 's statement above the only situation where T1/T2 would be deleted on disk is if they were overwritten by performing a new calculation, or if all of the scratch files are removed such as by calling `clean()`.